internal/controllers/addon/monitoring_federation_reconciler.go

```diff
@@ -242,7 +242,8 @@ func (r *monitoringFederationReconciler) ensureDeletionOfUnwantedMonitoringFeder
 	}
 
 	for _, serviceMonitor := range currentServiceMonitors {
-		if serviceMonitor.Name == wantedServiceMonitorName {
+		if serviceMonitor.Name == wantedServiceMonitorName ||
+			serviceMonitor.Name == getServiceMonitorName(addon.Name) {
```

Comment:
So does this mean that the service monitor/s created by the addon team can only have `getServiceMonitorName(addon.Name)` name?

Reply:
Nope that's not the case, unless they created their servicemonitors with labels denoting that that service-monitor is actually managed by the addon-operator.

Reply:
if you notice, the `currentServiceMonitors` variable doesn't iterate through all ServiceMonitors in the cluster but only those servicemonitors in the cluster who are managed by ADO (indicated by certain labels on it).
PS: checkout the `r.getOwnedServiceMonitorsViaCommonLabels` :)

Reply:
That being said, you had a good catch. That line of code `serviceMonitor.Name == getServiceMonitorName(addon.Name) ` has to be removed as it is although not breaking anything, it is not serving anything too.